test/gui/shared/steps/steps.py

```diff
@@ -542,122 +436,62 @@ def step(context, receiver, resource):
-    mouseClick(
-        waitForObject(names.oCC_ShareLinkWidget_lineEdit_password_QLineEdit),
-        0,
-        0,
-        Qt.NoModifier,
-        Qt.LeftButton,
-    )
-    type(waitForObject(names.oCC_ShareLinkWidget_lineEdit_password_QLineEdit), password)
-    clickButton(waitForObject(names.oCC_ShareLinkWidget_createShareButton_QPushButton))
-    waitFor(
-        lambda: (
-            findObject(names.linkShares_0_0_QModelIndex).displayText == "Public link"
-        )
-    )
+    openSharingDialog(context, resource)
+    publicLinkDialog = PublicLinkDialog()
+    publicLinkDialog.openPublicLinkDialog()
+    publicLinkDialog.createPublicLink(context, resource, password, '', '')
 
 
 @When('the user edits the public link named "|any|" of file "|any|" changing following')
 def step(context, publicLinkName, resource):
-    test.compare(
-        str(waitForObjectExists(names.sharingDialog_label_name_QLabel).text), resource
-    )
-    test.compare(
-        str(waitForObjectExists(names.linkShares_0_0_QModelIndex).text), publicLinkName
-    )
-    expDate = []
-    for row in context.table:
-        if row[0] == 'expireDate':
-            expDate = datetime.datetime.strptime(row[1], '%Y-%m-%d')
-    expYear = expDate.year - 2000
-    mouseClick(
-        waitForObject(names.oCC_ShareLinkWidget_qt_spinbox_lineedit_QLineEdit),
-        0,
-        0,
-        Qt.NoModifier,
-        Qt.LeftButton,
-    )
-    nativeType("<Delete>")
-    nativeType("<Delete>")
-    nativeType(expDate.month)
-    nativeType(expDate.day)
-    nativeType(expYear)
-    nativeType("<Return>")
-    testSettings.silentVerifications = True
-    waitFor(lambda: (test.xvp("publicLinkExpirationProgressIndicatorInvisible")))
-    waitFor(lambda: (test.vp("publicLinkExpirationProgressIndicatorInvisible")))
-    testSettings.silentVerifications = False
-    test.compare(
-        str(
-            waitForObjectExists(
-                names.oCC_ShareLinkWidget_qt_spinbox_lineedit_QLineEdit
-            ).displayText
-        ),
-        str(expDate.month) + "/" + str(expDate.day) + "/" + str(expYear),
-    )
+    publicLinkDialog = PublicLinkDialog()
+    publicLinkDialog.setExpirationDate(context, publicLinkName, resource)
 
 
-@When(
-    'the user creates a new public link with permissions "|any|" for folder "|any|" without password using the client-UI'
-)
-def step(context, permissions, resource):
-    resource = sanitizePath(substituteInLineCodes(context, resource))
-    openPublicLinkDialog(context, resource)
+def getRadioObjectForPermssion(permissions):
     radioObjectName = ''
     if permissions == 'Download / View':
         radioObjectName = names.oCC_ShareLinkWidget_radio_readOnly_QRadioButton
     elif permissions == 'Download / View / Edit':
         radioObjectName = names.oCC_ShareLinkWidget_radio_readWrite_QRadioButton
     elif permissions == 'Upload only (File Drop)':
         radioObjectName = names.oCC_ShareLinkWidget_radio_uploadOnly_QRadioButton
```

Comment:
Maybe this can be done in page objects. since it's related to the selectors it makes more sense to do it there.

Reply:
Done